kernel/syscall.c

```diff
@@ -2838,6 +2838,28 @@ long myst_syscall_set_robust_list(myst_robust_list_head_t* head, size_t len)
     return ret;
 }
 
+long myst_syscall_arch_prctl(int code, unsigned long* addr)
+{
+    long ret = 0;
+
+    if (code == ARCH_GET_FS)
+    {
+        unsigned long* fs = myst_get_fsbase();
+        *addr = *fs;
```

Comment:
This should be:
```
*addr = (unsigned long)fs;
```

Otherwise you are storing the first word that fsbase points to into address rather than the pointer itself.

Reply:
Thanks for catching this! Fixed